### DISCUSSION TITLE
Link the emails automatically to email history

### DISCUSSION THREAD

**Discussion by Ilamparithi-Tamizhanangu** [YetiForceCompany/YetiForceCRM#15815] [How to] [answered]
Hi,

I synchronised my gmail account with mail scanner and i can able to view the mails in My mailbox section.

When i check the Email History of the particular contact, it remains empty. 
![image](https://user-images.githubusercontent.com/96602176/160062794-e9bbf5fc-d958-43e0-bc2d-9bb7f8dda0f2.png)

I need to manually link the each mail to its contacts. After manually added to the related record, it was displayed in Email history. 
![image](https://user-images.githubusercontent.com/96602176/160062886-c6df3eb4-4a7e-400e-9bda-2a21fce45fa1.png)

Could you please guide me how to add all the existing mails to related contacts and link all new mails to related contacts. 

Thanks in advance. 

with regards,
Rajasekaran.

**Comment by rskrzypczak** [comment, marked as answer]
Have you properly configured the mail scanner?
You must indicate in which modules and in which e-mail fields the scanner should search for records.
![image](https://user-images.githubusercontent.com/10221999/160066319-031c4166-2a21-4899-8144-6d882fc6df97.png)

**Reply by Ilamparithi-Tamizhanangu** [reply]
Hi,

Yes, i have selected the action as "Relate with contacts"
![image](https://user-images.githubusercontent.com/96602176/160067161-d8aedeb6-e729-44cd-bc26-1762fc58b6c2.png)

and it need to search in contacts primary and secondary email fields.
![image](https://user-images.githubusercontent.com/96602176/160067254-843df8bf-7b9c-4afe-8e80-551756a4286a.png)


Do i need to change any other parameters ? 

Thanks & regards,
rajasekaran

**Reply by rskrzypczak** [reply]
The binding action itself is pointless, you always have to add _Create email massage_
![image](https://user-images.githubusercontent.com/10221999/160067916-8802c874-8da5-478c-ab56-3f9a77a81380.png)

**Reply by Ilamparithi-Tamizhanangu** [reply]
Hi,

Thanks for the solution. Now mails are started to link with contacts. 

Some emails I received one month earlier are linked with suitable contacts. But some emails, i received last week are not yet linked to the contacts. 

Does it follow any specific order to link the old emails? 
If I create any new contact and does it link old emails to the newly created contacts? 

Thanks.

**Reply by rskrzypczak** [reply]
Nothing happens by itself, but you can try to reset relationship.
![image](https://user-images.githubusercontent.com/10221999/160072939-36ec40c3-850e-4e0a-97a1-728c3df6a67c.png)

**Reply by Ilamparithi-Tamizhanangu** [reply]
hi,

Thanks for your answers.  It solved my queries. :)
